### PR TITLE
X5C Provisioner: add example on how to restrict issued cert to signing cert Subject

### PR DESCRIPTION
**Pull request by Slamdunk**
See https://github.com/smallstep/certificates/pull/827

### CODE REVIEW

File: src/pages/docs/step-ca/provisioners.mdx [hunk 1]
```diff
@@ -678,6 +678,30 @@ see the [claims](#claims) section for all the options.
 
 <Footnote id="star4" marker="*">Optional</Footnote>
 
+<Alert severity="warning">
+  <div> 
+    By default every valid certificate can issue through `X5C` provisioner new certificates
+    for <strong>any</strong> Subject or Principal. You can force allowed subject to
+    signing certificate's one within templates by accessing the `.AuthorizationCrt` variable.
```
Comment (tashian) [inline src/pages/docs/step-ca/provisioners.mdx]:
```
By default, the `X5C` provisioner will issue a certificates for <strong>any</strong> Subject names.
If you want to limit or modify the subject names this provisioner will issue,
you can use a <a href="/docs/step-ca/templates">certificate template</a>.
```

File: src/pages/docs/step-ca/provisioners.mdx [hunk 2]
```diff
@@ -678,6 +678,30 @@ see the [claims](#claims) section for all the options.
...
+    
+    For example, given a certificate issued by the OIDC provider to `foo@bar.com`, set the
+    `X5C` ssh template as follow to allow SSH certificates only for `foo@bar.com` and `foo`:
```
Comment (tashian) [inline src/pages/docs/step-ca/provisioners.mdx]:
```
For example, say your users will use the X5C provisioner to exchange an X.509 client certificate for an SSH user certificate.
The X.509 client certificate has `alice@example.com` as its subject,
and you want to ensure the SSH certificate has both `alice` and `alice@example.com` as principals.
The following template will make that scenario possible:
```

File: src/pages/docs/step-ca/provisioners.mdx [hunk 3]
```diff
@@ -678,6 +678,30 @@ see the [claims](#claims) section for all the options.
...
+    
+    ```json
+    {
+        "type": {{ toJson .Type }},
+        "keyId": {{ toJson .AuthorizationCrt.Subject.CommonName }},
+        "principals": [
+            {{ replace "@bar.com" "" .AuthorizationCrt.Subject.CommonName | toJson }},
```
Comment (tashian) [inline src/pages/docs/step-ca/provisioners.mdx]:
```
{{ replace "@example.com" "" .AuthorizationCrt.Subject.CommonName | toJson }},
```

**Review by tashian — CHANGES_REQUESTED** [review]
Hi @Slamdunk, thank you for the PR on this.
I've suggested some specific changes.

Also, please update this PR to follow the convention of [semantic linefeeds](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).

Thanks!

**Comment by Slamdunk** [comment]
> I've suggested some specific changes.

Implemented, thank you



> Also, please update this PR to follow the convention of [semantic linefeeds](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).

As a non-native English speaker, honestly I have no idea how to adapt to it.
The examples in the link provided look exactly the same to me :shrug:

**Comment by tashian** [comment]
lgtm. Thanks @Slamdunk!

**Pull request merged**